### PR TITLE
Custom URLs for OpenAI API  and ollama servers 

### PR DESCRIPTION
**Pull request by d-popov**
Added the possibility to use custom LLM with OpenAI library by passing baseURL.
Also added option to configure ollama server when it is not running locally.

### CODE REVIEW

**Comment by mufeedvh** [comment]
Hey @d-popov, this looks good to me, would you be able to resolve the merge conflicts? Will merge this after resolving them! Thanks! ❤️ 🙌

File: src/llm/ollama_client.py [hunk 1]
```diff
@@ -2,22 +2,45 @@
...
+    @classmethod
+    def list_models(self):
+        # if ollama_base is None - use ollama, else use the provided ollama_base by creating new client 
+        if self._ollama_client is None:
+            self.configure_ollama_client()
+    
         try:
-            return ollama.list()["models"]
+            models = self._ollama_client.list()["models"]
+            Logger().debug(f"Using Ollama server is serving theese models: {models}")
```
Comment (GustavHenning) [inline src/llm/ollama_client.py:29]: Typo: `theese`

**Comment by ARajgor** [comment]
#24

**Comment by Israel-Laguan** [comment]
I think this needs an update on the documentation, please consider adding instructions on the README

**Comment by d-popov** [comment]
for review

**Comment by ARajgor** [comment]
can you revert the Ollama file and just update the openai file with just one parameter? fetch the latest commits.

**Comment by d-popov** [comment]
@ARajgor I see the functionality is already on master. File reverted.

**Comment by ARajgor** [comment]
> @ARajgor I see the functionality is already on master. File reverted. I am sorry i don't understand your comment on openai_client.py

take the current openai_client file and just add base_url parameter.

**Comment by d-popov** [comment]
@ARajgor done

**Comment by ARajgor** [comment]
also, revert ollama file. the current file works perfectly. also add openai URL parameter in sample.config.file

File: src/llm/openai_client.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
-from openai import OpenAI
+from openai import OpenAI as OAI
```
Comment (d-popov) [inline src/llm/openai_client.py:1]: I've added the alias, as otherwise the constructor fails

**Comment by d-popov** [comment]
oh, well...

**Pull request closed without merging**